src/maidsafe/vault/mpid_manager/database.cc

```diff
@@ -59,7 +57,183 @@ MpidManagerDataBase::~MpidManagerDataBase() {
+        if (check_holder_result.new_holder == NodeId())
+          continue;
+        LOG(kVerbose) << "Db::GetTransferInfo having new holder " << check_holder_result.new_holder;
+        groups_to_be_transferred.push_back(std::make_pair(check_holder_result.new_holder,
+                                                          group_name));
+      } else {
+  //      VLOG(VisualiserAction::kRemoveAccount, key.name);
+        LOG(kInfo) << "Db::GetTransferInfo current node is not in the closest of account "
+                   << HexSubstr(group_name);
+        groups_to_be_removed.push_back(group_name);
+        // empty NodeId indicates removing from local
+        groups_to_be_transferred.push_back(std::make_pair(NodeId(), group_name));
+      }
+    }
+  }
+  for (const auto& group_name : groups_to_be_removed)
+    DeleteGroup(group_name);
+
+  MpidManager::TransferInfo transfer_info;
+  for (const auto& transfer_entry : groups_to_be_transferred) {
+    std::string query("SELECT Chunk_Name from MpidManagerAccounts WHERE MPID=?");
+    sqlite::Statement statement{*db_, query};
+    statement.BindText(1, transfer_entry.second);
+    while (statement.Step() == sqlite::StepResult::kSqliteRow)
+      PutIntoTransferInfo(transfer_entry.first, transfer_entry.second,
+                          statement.ColumnText(0), transfer_info);
+  }
+  return transfer_info;
+}
+
+void MpidManagerDatabase::PutIntoTransferInfo(const NodeId& new_holder,
+                                              const std::string& group_name_string,
+                                              const std::string& key_string,
+                                              MpidManager::TransferInfo& transfer_info) {
+  MpidManager::GroupName group_name(ComposeGroupName(group_name_string));
+  MpidManager::MessageKey key(ComposeKey(key_string));
+  auto found_itr = transfer_info.find(new_holder);
+  if (found_itr != transfer_info.end()) {  // append
+    LOG(kInfo) << "Db::GetTransferInfo add into transfering account "
+               << HexSubstr(group_name_string) << " to " << new_holder;
+    found_itr->second.push_back(std::make_pair(group_name, key));
+  } else {  // create
+    LOG(kInfo) << "Db::GetTransferInfo create transfering account "
+               << HexSubstr(group_name_string) << " to " << new_holder;
+    std::vector<MpidManager::GKPair> group_key_vector;
+    group_key_vector.push_back(std::make_pair(group_name, key));
+    transfer_info.insert(std::make_pair(new_holder, std::move(group_key_vector)));
+  }
+}
+
+bool MpidManagerDatabase::HasGroup(const MpidManager::GroupName& mpid) {
+  if (!db_)
+    BOOST_THROW_EXCEPTION(MakeError(CommonErrors::db_not_present));
+  std::string query("SELECT DISTINCT MPID from MpidManagerAccounts WHERE MPID=?");
+  sqlite::Statement statement{*db_, query};
+  auto group_name(EncodeGroupName(mpid));
+  statement.BindText(1, group_name);
+  while (statement.Step() == sqlite::StepResult::kSqliteRow)
```

Comment:
Couldn't this be reduced to

    return (statement.Step() == sqlite::StepResult::kSqliteRow);